trie/trie.go

```diff
@@ -865,29 +886,44 @@ func (t *Trie) ReviveTrie(proof []*MPTProofNub) (successNubs []*MPTProofNub) {
 
 func (t *Trie) TryRevive(proof []*MPTProofNub) (successNubs []*MPTProofNub, err error) {
 
+	// Revive trie with each proof nub
 	for _, nub := range proof {
-		newNode, didResolve, err := t.tryRevive(t.root, nub.n1PrefixKey, *nub)
+		// TODO (asyukii): Check if MPT Root is a RootNode object
+		// Get root node
+		// var root node
+		// if rt, ok := t.root.(RootNode); ok {
+		// 	root = rt
+		// } else {
+		// 	root = t.root
+		// }
+		root := t.root
+		path := []byte{}
+		newNode, didResolve, err := t.tryRevive(root, nub.n1PrefixKey, *nub, path, false) // TODO (asyukii): change isExpired to false
 		if didResolve && err == nil {
 			successNubs = append(successNubs, nub)
 			t.root = newNode
 		}
 	}
 
+	// If no nubs were successful, return error
 	if len(successNubs) == 0 && len(proof) != 0 {
 		return successNubs, fmt.Errorf("all nubs failed to revive trie")
 	}
 
 	return successNubs, nil
 }
 
-func (t *Trie) tryRevive(n node, key []byte, nub MPTProofNub) (node, bool, error) {
-	if len(key) == 0 {
+func (t *Trie) tryRevive(n node, key []byte, nub MPTProofNub, path []byte, isExpired bool) (node, bool, error) {
```

Comment:
Where is the parameter 'path' used?

Reply:
'path' is used only in ExpiredNodeError, returning the partial path to the expired node